### PR TITLE
feat(HACBS-2290): pass snapshot, releaseStrategy, and releasePlan obj ref

### PR DESCRIPTION
**Pull request by johnbieren**

### CODE REVIEW

**Comment by johnbieren** [comment]
Marking this as do not merge until I have new versions of the release pipelines ready for this change

File: controllers/release/adapter.go [hunk 1]
```diff
@@ -302,13 +302,13 @@ func (a *Adapter) EnsureReleaseProcessingIsTracked() (reconciler.OperationResult
 // PipelineRun.
 func (a *Adapter) createReleasePipelineRun(resources *loader.ProcessingResources) (*v1beta1.PipelineRun, error) {
 	pipelineRun := tekton.NewReleasePipelineRun("release-pipelinerun", resources.ReleaseStrategy.Namespace).
-		WithObjectReferences(a.release, resources.ReleasePlanAdmission).
+		WithObjectReferences(a.release, resources.ReleasePlan,
+			resources.ReleasePlanAdmission, resources.ReleaseStrategy, resources.Snapshot).
```
Comment (mmalina) [inline controllers/release/adapter.go:306]: This also adds the `ReleaseStrategy`, right? Can you change the commit message / PR description to that effect?

**Review by mmalina — CHANGES_REQUESTED** [review]

File: controllers/release/adapter.go (hunk 1 above)
Comment (johnbieren) [reply controllers/release/adapter.go:306]: done

Comment (mmalina) [reply controllers/release/adapter.go:306]: Thanks.

File: loader/loader.go [hunk 2]
```diff
@@ -304,6 +305,11 @@ func (l *loader) GetProcessingResources(ctx context.Context, cli client.Client,
 	var err error
 	resources := &ProcessingResources{}
 
+	resources.ReleasePlan, err = l.GetReleasePlan(ctx, cli, release)
```
[the remark this answers is not in the document]
Comment (johnbieren) [reply loader/loader.go:308]: good catch, fixed

[the remark this answers is not in the document]
Comment (mmalina) [reply loader/loader.go:308]: Cool, thanks.

**Review by mmalina — APPROVED** [review]

**Review by kasemAlem — APPROVED** [review]

**Pull request merged**